File: find_dupes.js
const fs = require('fs');

function findDuplicateAttrs(content) {
    const tagRegex = /<([A-Z][a-zA-Z0-9.]+|[a-z][a-z0-9-]*)\s+([^>]+)\/?>/gs;
    const results = [];
    let match;

    while ((match = tagRegex.exec(content)) !== null) {
        const tagName = match[1];
        const attrsStr = match[2];

        // Match attribute names: word characters followed by = or space/end of string
        const attrNameRegex = /([a-zA-Z0-9-]+)(?:\s*=\s*(?:\{[^\}]+\}|"[^"]*"|'[^']*')|\s+|$)/g;
        const attrNames = [];
        let attrMatch;
        while ((attrMatch = attrNameRegex.exec(attrsStr)) !== null) {
            attrNames.push(attrMatch[1]);
        }

        const seen = new Set();
        const duplicates = new Set();
        for (const name of attrNames) {
            if (seen.has(name)) {
                duplicates.add(name);
            }
            seen.add(name);
        }

        if (duplicates.size > 0) {
            results.push({
                tag: tagName,
                duplicates: Array.from(duplicates),
                full: attrsStr.trim()
            });
        }
    }
    return results;
}

const filePath = 'src/pages/agent/AgentProfile.tsx';
const content = fs.readFileSync(filePath, 'utf8');
const duplicates = findDuplicateAttrs(content);

if (duplicates.length > 0) {
    duplicates.forEach(d => {
        console.log(`Duplicate attributes [${d.duplicates.join(', ')}] in tag <${d.tag}>`);
        console.log(`Full attributes string: ${d.full}`);
        console.log('-'.repeat(20));
    });
} else {
    console.log('No duplicate attributes found.');
}

File: find_dupes.py
import re
import sys

def find_duplicate_attrs(content):
    # Regex to find JSX tags and their attributes
    # This is a simple regex and might not handle all edge cases (like strings with >)
    tags = re.findall(r'<([A-Z][a-zA-Z0-9.]+|[a-z][a-z0-9-]*)\s+([^>]+)/?>', content, re.DOTALL)
    
    results = []
    for tag_name, attrs_str in tags:
        # Regex to find attribute names
        # Handles name="value", name={value}, and boolean name
        attr_names = re.findall(r'([a-zA-Z0-9-]+)(?:\s*=\s*(?:\{[^\}]+\}|"[^"]*"|\'[^\']*\')|\s+|$)', attrs_str)
        
        seen = set()
        duplicates = set()
        for name in attr_names:
            if name in seen:
                duplicates.add(name)
            seen.add(name)
            
        if duplicates:
            results.append((tag_name, list(duplicates), attrs_str.strip()))
            
    return results

if __name__ == "__main__":
    file_path = "src/pages/agent/AgentProfile.tsx"
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()
    
    duplicates = find_duplicate_attrs(content)
    if duplicates:
        for tag, dups, full in duplicates:
            print(f"Duplicate attributes {dups} in tag <{tag}>:")
            print(f"Full attributes string: {full}")
            print("-" * 20)
    else:
        print("No duplicate attributes found.")
